work/worker.go

```diff
@@ -1010,26 +1071,40 @@ func (env *Task) commitBundleTransaction(bundle *builder.Bundle, bc BlockChain,
 			return kerrors.ErrTxGeneration, nil, nil
 		}
 
-		env.state.SetTxContext(tx.Hash(), common.Hash{}, env.tcount)
-		receipt, _, err := bc.ApplyTransaction(env.config, &nodeAddr, env.state, env.header, tx, &env.header.GasUsed, vmConfig)
-		// Bundled tx will be rejected with any receipt.Status other than success.
-		// There may be cases where a revert occurs within the EVM, which could result in an attack on a tx sender in an already executed bundle.
-		if err != nil || receipt.Status != types.ReceiptStatusSuccessful {
-			if err != vm.ErrInsufficientBalance && err != vm.ErrTotalTimeLimitReached {
-				markAllTxUnexecutable()
-			}
-			receiptStatus := ""
-			if receipt != nil {
-				receiptStatus = strconv.FormatUint(uint64(receipt.Status), 10)
-			}
-			logger.Warn("ApplyTransaction error, restoring env",
-				"blockNum", env.header.Number.String(), "txHash", tx.Hash().String(),
-				"error", err, "receiptStatus", receiptStatus,
-			)
-			restoreEnv()
-			if err == nil {
-				err = kerrors.ErrRevertedBundleByVmErr
+		executTx := func() (error, *types.Receipt) {
+			env.state.SetTxContext(tx.Hash(), common.Hash{}, env.tcount)
+			receipt, _, err := bc.ApplyTransaction(env.config, &nodeAddr, env.state, env.header, tx, &env.header.GasUsed, vmConfig)
+			// Bundled tx will be rejected with any receipt.Status other than success.
+			// There may be cases where a revert occurs within the EVM, which could result in an attack on a tx sender in an already executed bundle.
+			if err != nil || receipt.Status != types.ReceiptStatusSuccessful {
+				if err != vm.ErrInsufficientBalance && err != vm.ErrTotalTimeLimitReached {
+					markAllTxUnexecutable()
+				}
+				receiptStatus := ""
+				if receipt != nil {
+					receiptStatus = strconv.FormatUint(uint64(receipt.Status), 10)
+				}
+				logger.Warn("ApplyTransaction error, restoring env",
+					"blockNum", env.header.Number.String(), "txHash", tx.Hash().String(),
+					"error", err, "receiptStatus", receiptStatus,
+				)
+				restoreEnv()
```

Comment:
In case of tx revert, is this the expected behavior?
(1) the reverting blobTx is mined in block
(2) `header.BlobGasUsed` is not increased
(3) Sidecar is stored

(2) and (3) seems to be contradictory.

Also, it seems KIP-279 does not state what to do in such case.